diff --git a/module/os/operation_siren.py b/module/os/operation_siren.py
index b8dbdd1..d254c59 100644
--- a/module/os/operation_siren.py
+++ b/module/os/operation_siren.py
@@ -379,7 +379,8 @@ class OperationSiren(OSMap):
                     self.os_order_execute(
                         recon_scan=False,
                         submarine_call=self.config.OpsiFleet_Submarine)
-                    self.run_auto_search()
+                    # self.run_auto_search()
+                    self.run_strategic_search()
                     self.handle_after_auto_search()
                     self.config.check_task_switch()
             else:
