diff --git a/module/config/argument/argument.yaml b/module/config/argument/argument.yaml
index 7e17ccd..0b83b28 100644
--- a/module/config/argument/argument.yaml
+++ b/module/config/argument/argument.yaml
@@ -688,6 +691,9 @@ OpsiExplore:
   SpecialRadar: false
   ForceRun: false
   LastZone: 0
+  HazardLevel:
+    value: 5
+    option: [0, 1, 2, 3, 4, 5]
 OpsiShop:
   PresetFilter:
     value: max_benefit_meta
diff --git a/module/os/operation_siren.py b/module/os/operation_siren.py
index b8dbdd1..d254c59 100644
--- a/module/os/operation_siren.py
+++ b/module/os/operation_siren.py
@@ -491,8 +492,9 @@ class OperationSiren(OSMap):
                 self.config.task_call('OpsiHazard1Leveling', force_call=False)
             self.config.task_stop()
 
-        logger.hr('OS explore', level=1)
-        order = [int(f.strip(' \t\r\n')) for f in self.config.OS_EXPLORE_FILTER.split('>')]
+        logger.hr(f'OS explore hazard_level option = {self.config.OpsiExplore_HazardLevel}', level=1)
+        filter = ' '.join(self.config.OS_EXPLORE_FILTER.split('\n\n')[:self.config.OpsiExplore_HazardLevel+1])
+        order = [int(f.strip(' \t\r\n')) for f in filter.split('>')]
         # Convert user input
         try:
             last_zone = self.name_to_zone(self.config.OpsiExplore_LastZone).zone_id
