diff --git a/module/handler/info_handler.py b/module/handler/info_handler.py
index bb64d52..78fc4a4 100644
--- a/module/handler/info_handler.py
+++ b/module/handler/info_handler.py
@@ -97,12 +97,12 @@ class InfoHandler(ModuleBase):
         if self.appear(POPUP_CANCEL, offset=offset) \
                 and self.appear(POPUP_CONFIRM, offset=offset, interval=interval):
             POPUP_CONFIRM.name = POPUP_CONFIRM.name + '_' + name
-            self.device.click(POPUP_CONFIRM)
+            self.device.click(POPUP_CONFIRM, control_check=False)
             POPUP_CONFIRM.name = POPUP_CONFIRM.name[:-len(name) - 1]
             return True
         if self.appear(POPUP_CONFIRM_WHITE, offset=offset, interval=interval):
             POPUP_CONFIRM_WHITE.name = POPUP_CONFIRM_WHITE.name + '_' + name
-            self.device.click(POPUP_CONFIRM_WHITE)
+            self.device.click(POPUP_CONFIRM_WHITE, control_check=False)
             POPUP_CONFIRM_WHITE.name = POPUP_CONFIRM_WHITE.name[:-len(name) - 1]
             return True
         return False
diff --git a/module/os/map.py b/module/os/map.py
index 078ef35..1a6db92 100644
--- a/module/os/map.py
+++ b/module/os/map.py
@@ -831,10 +832,10 @@ class OSMap(OSFleet, Map, GlobeCamera, StrategicSearchHandler):
         if 'is_scanning_device' not in self._solved_map_event and grids and grids[0].is_scanning_device:
             grid = grids[0]
             logger.info(f'Found scanning device on {grid}')
-            if self.is_in_task_cl1_leveling:
-                logger.info('In CL1 leveling, mark scanning device as solved')
-                self._solved_map_event.add('is_scanning_device')
-                return True
+            # if self.is_in_task_cl1_leveling:
+            #     logger.info('In CL1 leveling, mark scanning device as solved')
+            #     self._solved_map_event.add('is_scanning_device')
+            #     return True
 
             self.device.click(grid)
             with self.config.temporary(STORY_ALLOW_SKIP=False):
